File: Week_03/78/subsets.go
package main

func subsets(nums []int) [][]int {
	res := make([][]int, 0)
	var dfs func(arr []int, idx int)
	dfs = func(arr []int, idx int) {
		res = append(res, append([]int{}, arr...))
		for i := idx; i < len(nums); i++ {
			dfs(append(arr, nums[i]), i+1)
		}
	}
	dfs([]int{}, 0)
	return res
}

File: Week_03/78/subsets.py
from typing import List


class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []

        def dfs(arr: List[int], idx: int):
            res.append(arr.copy())
            for i in range(idx, len(nums)):
                dfs(arr + [i])

        dfs([], 0)
        return res
